network/builder/src/builder.rs

```diff
@@ -144,19 +128,32 @@ impl NetworkBuilder {
 
         let pub_key = authentication_mode.public_key();
 
-        let mut network_builder = NetworkBuilder::new(
-            runtime.handle().clone(),
-            chain_id.clone(),
+        let network_context = Arc::new(NetworkContext::new(
             config.network_id.clone(),
             role,
             peer_id,
+        ));
+
+        let trusted_peers = if config.mutual_authentication && role == RoleType::FullNode {
+            Arc::new(RwLock::new(config.network_peers.clone()))
+        } else {
+            Arc::new(RwLock::new(HashMap::new()))
+        };
```

Comment:
I guess this looks a little out of date....

frankly, our support for mutual authentication except for the validator network is not tested or documented... I am out until next Thursday, flagging @phlip9 / @gregnazario to discuss internally if we want to keep this or kill this, since it is actually causing us to push around "tech debt"

Reply:
This and the following both expose some unfortunate tech debt pieces that I'm continuing to push around.

At least the tech debt is more visible now.